qiskit/dagcircuit/dagcircuit.py

```diff
@@ -1393,6 +1393,28 @@ def substitute_node(self, node, op, inplace=False):
             self._decrement_op(node.op)
         return new_node
 
+    def weakly_connected_components(self) -> List["DAGCircuit"]:
+        """Decompose the dag circuit into its weakly connected components."""
```

Comment:
I don't want to get too derailed on bikeshedding the name, but I'd perhaps prefer to go for something a bit more quantum adjacent than graph terminology here.  Maybe some variant on `separable_components` or `separable_circuits`?

Reply:
Changed to `separable_circuits`